src/realm/object-store/c_api/object.cpp

```diff
@@ -157,6 +157,24 @@ RLM_API const void* _realm_object_get_native_ptr(realm_object_t* obj)
     return static_cast<const Object*>(obj);
 }
 
+RLM_API realm_object_t* realm_object_freeze(const realm_object_t* live_object, const realm_t* frozen_realm)
+{
+    return wrap_err([&]() {
+        const auto& realm = *frozen_realm;
+        auto frozen_object = live_object->freeze(realm);
+        return new realm_object_t{std::move(frozen_object)};
```

Comment:
Similar question here - what happens if the object doesn't exist in the frozen Realm?

Reply:
I don't think this would be an actual use case. From Kotlin SDK's side (at least at the moment) we would only do this if we have an internal error. Given that the underlying object store API doesn't allow returning null/optionals I wouldn't expect other SDK's to rely on this either. I would consider fixing the behavior of existing freeze's for another PR.